sklbench/utils/common.py

```diff
@@ -0,0 +1,289 @@
+# ===============================================================================
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+from pprint import pformat
+import hashlib
+import re
+import json
+from typing import Any, Dict, List, Union, Tuple
+from .custom_types import JsonTypesUnion, ModuleContentMap, BenchCase, Numeric
+import subprocess as sp
+import inspect
+import importlib
+from shutil import get_terminal_size
+from copy import deepcopy
+
+
+HASH_LIMIT = 6
+PFORMAT_INDENT = 4
+BCOLORS = {
+    'HEADER': '\033[95m',
+    'OKBLUE': '\033[94m',
+    'OKCYAN': '\033[96m',
+    'OKGREEN': '\033[92m',
+    'WARNING': '\033[93m',
+    'FAIL': '\033[91m',
+    'ENDC': '\033[0m',
+    'BOLD': '\033[1m',
+    'UNDERLINE': '\033[4m'
+}
+
+
+def custom_format(
+        input_obj: Any,
+        bcolor: str | None = None,

```

Comment:
This union operator was introduced in Python 3.10 (https://docs.python.org/3/whatsnew/3.10.html)
This causes issues in older Python versions

https://peps.python.org/pep-0604/

I can't seem to suggest a commit but try:
Union[str, None] = None